### PR TITLE
Unregister the client

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -181,22 +181,13 @@ class App {
...
-    //   } else {
-    //     log.info('Current device properly removed from remote cozy.')
-    //     fs.remove(this.basePath, callback)
-    //   }
-    // })
+  removeRemote () {
+    if (!this.remote) {
+      this.instanciate()
+    }
+    this.remote.unregister()
```
Comment (sebn) [inline src/app.js]: Catch failures?

File: src/remote/cozy.js [hunk 2]
```diff
@@ -42,6 +42,7 @@ export default class RemoteCozy {
     })
 
     // Aliases:
+    this.unregister = this.client.auth.unregisterClient
```
Comment (sebn) [inline src/remote/cozy.js]:
```javascript
unregister: () => Promise<*>
```

Should fix flow.

File: src/app.js [hunk 3]
```diff
@@ -151,22 +143,18 @@ class App {
...
+      if (!this.remote) {
+        this.instanciate()
+      }
+      this.remote.unregister()
+      fs.removeSync(this.basePath)
+      log.info('Current device properly removed from remote cozy.')
+    } catch (err) {
+      log.error('An error occured while unregistering your device.')
+      log.error(err)
+    }
```
Comment (sebn) [inline src/app.js]: Won't this miss failures since `unregister()` is async?

Comment (nono) [inline src/app.js]: You are right, it's better to make `await` here

File: src/app.js [hunk 4]
```diff
@@ -143,17 +143,19 @@ class App {
 
   // Unregister current device from remote Cozy and then remove remote from
   // the config file
-  removeRemote () {
+  async removeRemote () {
     try {
       if (!this.remote) {
         this.instanciate()
       }
       this.remote.unregister()
```
Comment (sebn) [inline src/app.js]: `await`?

Comment (nono) [inline src/app.js]: Yeah, my mistake

**Pull request merged**